src/stores/useAuthStore.ts

```diff
@@ -6,14 +6,14 @@ export interface User {
   name: string;
 }
 
-export interface UserStoreState {
+export interface AuthStoreState {
   user: User | null;
   setUser(user: User | null): void;
   token: string | null;
   setToken(token: string | null): void;
 }
 
-export const useUserStore = create<UserStoreState>()(
+export const useAuthStore = create<AuthStoreState>()(
```

Comment:
I'd like to propose a different approach for structuring stores where we separate setters like `setUser` or `setToken` from the actual state `user` and `token` as outlined in this article from the zustand docs: https://zustand.docs.pmnd.rs/guides/practice-with-no-store-actions

Besides the small benefits described in the article, I've also found that when persisting the store to `localStorage` these function definitions do not serve any purpose and can also cause issues being not serializable.

Reply:
What do you think using this as a standard but also gathering everything from this file in a namespace in the barrel file for example:

```
export * as authStore from "./useAuthStore"; 
```

This approach allows us to achieve both objectives: separating concerns while avoiding multiple setters with similar names, as often seen in regular component states or sections. 

What do you think?

Reply:
Yes, sounds good!